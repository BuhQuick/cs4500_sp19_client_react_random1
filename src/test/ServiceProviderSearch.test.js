import React from 'react'
import renderer from 'react-test-renderer'
import mockService from "../mock/data/service.dto.mock";
import ServicesProviderSearch from "../components/serviceProviderSearch/ServiceProviderSearch"
import {shallow} from 'enzyme';

test('service provider search renders correctly search fields', () => {
    const testRenderer = renderer.create(
        <ServicesProviderSearch match={{params: {id: 123}}}/>)

    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();

    const testInstance = testRenderer.root;

    const twoRowContainers = testInstance.findAllByProps({className: 'row'});
    const twoInputs = testInstance.findAllByProps({className: 'form-control'});

    expect(twoRowContainers.length).toBe(2);
    expect(twoInputs.length).toBe(2);
});

describe('ServiceProviderSearch', () => {
    it('test load providers', () => {
        const component = shallow(<ServicesProviderSearch match={{ params: { id: 123 }}}/>);
        component.instance().setState({
            service: mockService
        });

        const providers = component.instance().generateProviders();
        component.instance().componentDidMount();

        expect(providers.length).toBe(3);
        expect(component.instance().state.service).toExist();
    })
});

