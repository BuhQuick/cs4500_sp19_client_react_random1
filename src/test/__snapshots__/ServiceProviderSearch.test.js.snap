// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`service provider search renders correctly search fields 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="input-group input-group-lg col-8"
    >
      <input
        className="form-control"
        placeholder="Search for providers"
        type="text"
      />
      <input
        className="form-control"
        placeholder="Zip Code"
        type="text"
      />
      <button
        className="btn btn-primary"
      >
        Search
      </button>
    </div>
    <div
      className="col-4 row"
    >
      <div
        className="col-11 text-right"
      >
        <a
          href="#"
        >
          Sign up
        </a>
      </div>
      <div
        className="col-1"
      >
        <a
          href="#"
        >
          Log in
        </a>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-3"
    />
    <div
      className="col-9"
    />
  </div>
</div>
`;
